kotlinx-coroutines-core/jvm/test/sync/RateLimiterTest.kt

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2016-2021 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.sync
+
+import kotlinx.coroutines.*
+import kotlinx.coroutines.test.*
+import kotlinx.coroutines.time.*
+import org.junit.*
+import org.junit.Test
+import org.junit.runner.*
+import org.junit.runners.*
+import org.junit.runners.Parameterized.*
+import kotlin.math.*
+import kotlin.test.*
+import kotlin.time.*
+
+@RunWith(Parameterized::class)
+@ExperimentalTime
+@ExperimentalThroughputLimiter
+internal class RateLimiterParamTest(private val eventsPerInterval: Int) {
+
+    companion object {
+
+        @JvmStatic
+        @Parameters(name = "{0} events per interval")
+        fun data(): Collection<Array<Int>> = listOf(1, 3, 10, 100, 1000).map { arrayOf(it) }
+    }
+
+    @Test
+    internal fun acquire(): Unit = runBlocking {
+        val delayer = Delayer()
+        val timeSource: NanoTimeSource = TestNanoTimeSource()
+        val rateLimiter = RateLimiterImpl(
+            eventsPerInterval = eventsPerInterval,
+            interval = Duration.seconds(1),
+            timeSource = timeSource,
+            delay = delayer::delay
+        )
+        (0..eventsPerInterval).forEach { delayMultiplier ->
+            delayer.reset()
+            val actualDelay = rateLimiter.acquire()
+            val delaySum = delayer.getDelay()
+            val expectedDelay: Long = (1000L / eventsPerInterval) * delayMultiplier
+            assertEquals(
+                expected = expectedDelay,
+                actual = actualDelay,
+                message = "Returned sleep time should be $expectedDelay ms"
```

Comment:
Assert failure would already include `expected` value, so there's no point in duplicating that in the message.

On the other hand, the `message = ...` parameter should clarify **the reason** the expected value should be like that.

It would make the failure message easier to understand, and it would help to review the test. For instance, I see here's some loop, however, it is not clear why the expected delay is like that.

Reply:
Thanks for your feedback @vlsi , but I think this PR has been ignored for more than a year now. I will not pay it any more mind unless they respond to this or the issues I've posted on their issue-tracker 🤷

Reply:
🤝  🤷‍♂️  😭